communication/src/coap_channel.cpp

```diff
@@ -124,15 +131,17 @@ ProtocolError CoAPMessageStore::receive(Message& msg, Channel& channel, system_t
 	if (msgtype==CoAPType::ACK || msgtype==CoAPType::RESET)
 	{
 		message_id_t id = msg.get_id();
-		if (msgtype==CoAPType::RESET) {
-			CoAPMessage* msg = from_id(id);
-			if (msg) {
-				msg->notify_delivered_nak();
+		CoAPMessage* coap_msg = from_id(id);
+		if (coap_msg)
+		{
+			g_coapLatencyMSec = time - coap_msg->get_send_time();
+			if (msgtype==CoAPType::RESET) {
+				coap_msg->notify_delivered_nak();
+				// a RESET indicates that the session is invalid.
+				// Currently the device never sends a RESET, but if it were to do that
+				// then we should track which direction we are sending
+				channel.command(Channel::DISCARD_SESSION, nullptr);
```

Comment:
This changes how `RST` messages are handled. Previously, the `DISCARD_SESSION` command would be called even if the original CoAP message was not found.

Reply:
@sergeuz great catch. thank u for checking. 
Changed! There are multiple ways to refactor this little logic- I just separated both the if statements. Please check.